services/ReservationService.go

```diff
@@ -0,0 +1,68 @@
+package services
+
+import (
+	"context"
+
+	"github.com/onego-project/onego/resources"
+)
+
+// ReservationService structure to manage reservation in OpenNebula virtual networks.
+type ReservationService struct {
+	Service
+}
+
+func (rs *ReservationService) reserve(ctx context.Context, parent resources.VirtualNetwork,
+	reservation resources.Reservation) (*resources.VirtualNetwork, error) {
+	vnID, err := parent.ID()
+	if err != nil {
+		return nil, err
+	}
+
+	reservationStr, err := resources.RenderInterfaceToXMLString(reservation)
+	if err != nil {
+		return nil, err
+	}
+
+	resArr, err := rs.call(ctx, "one.vn.reserve", vnID, reservationStr)
+	if err != nil {
+		return nil, err
+	}
+
+	vns := &VirtualNetworkService{Service: rs.Service}
+
+	return vns.RetrieveInfo(ctx, int(resArr[resultIndex].ResultInt()))
+}
+
+// Reserve reserves network addresses.
+func (rs *ReservationService) Reserve(ctx context.Context, parent resources.VirtualNetwork,
```

Comment:
Wouldn't it be better to get rid of this method and make `reserve` public since it's just passing arguments?